From fac21ad662a14a4f901b3365a25749b516918554 Mon Sep 17 00:00:00 2001
From: Igor Gnatenko <ignatenkobrain@fedoraproject.org>
Date: Mon, 24 Jun 2019 20:27:20 +0200
Subject: [PATCH] macros: Do not use awk's inplace feature

gawk 5.x changed variable name to `inplace::suffix` but gawk 4.x does
not like that name so we can't set both.

Signed-off-by: Igor Gnatenko <ignatenkobrain@fedoraproject.org>
---
 data/macros.cargo | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/data/macros.cargo b/data/macros.cargo
index 1182850..5066587 100644
--- a/data/macros.cargo
+++ b/data/macros.cargo
@@ -89,7 +89,8 @@ if %__cargo_is_lib; then                                            \
 # Drop all dependency/features information                          \
 # so that cargo doesn't fail resolving dependencies:                \
 # https://github.com/rust-lang/cargo/pull/6729                      \
-  %{__awk} -i inplace -v INPLACE_SUFFIX=.deps '/^\\\[((.+\\\.)?((dev|build)-)?dependencies|features)/{f=1;next} /^\\\[/{f=0}; !f' Cargo.toml \
+  mv Cargo.toml{,.deps}                                             \
+  awk '/^\\\[((.+\\\.)?((dev|build)-)?dependencies|features)/{f=1;next} /^\\\[/{f=0}; !f' Cargo.toml.deps > Cargo.toml \
   %{__cargo} package -l | xargs -d '\\\n' %{__cp} --parents -a -t $REG_DIR \
   %{__mv} Cargo.toml{.deps,}                                        \
   %{__cp} -a Cargo.toml $REG_DIR/Cargo.toml                         \
-- 
2.22.0

